### DISCUSSION TITLE
./duckdb < dml_statements.sql file1.csv file2.csv

### DISCUSSION THREAD

**Discussion by jimdickson007** [duckdb/duckdb#13935] [Q&A]
I want to call duck and run sql script = dml_statements.sql 
Within dml_statements.sql, I want to set variable1 = file1.csv & variable2 = file2.csv
( then I will read_csv those files )
Is this possible ? Is there a better way ?

My shell knowledge is ancient but suspect I am try to pass shell arguments $1 $1 into sql script.

As a bonus (and likely upcoming 2nd question), I would also like to generate and capture checksum/hash function of both files.

**Comment by Tishj** [comment]
https://duckdb.org/docs/sql/statements/set_variable.html

**Comment by Tishj** [comment]
Also https://duckdb.org/docs/api/cli/overview.html#reading-environment-variables

**Reply by jimdickson007** [reply]
ah - thank looks super useful, thanks
update 2024-09-29: Had problems with reading env variables in Windows - see https://github.com/duckdb/duckdb/discussions/14161
